demo.js

```diff
@@ -1,12 +1,61 @@
 import DanceParty from './src/p5.dance';
 import jazzy_beats from './metadata/jazzy_beats';
+const DanceAPI = require('./src/api');
+const interpreted = require('raw-loader!./src/p5.dance.interpreted.js');
+
+/**
+ * Inject p5.dance.interpreted code directly (this is only acceptable because we
+ * don't have untrusted user input in our dev env). Also attach all functionNames
+ * to the global namespace so that they can be called from the console, making
+ * for easier testing
+ */
+function injectInterpreted(api) {
+  // This list is not exhaustive, and more can/should be added as needed
+  const functionNames = [
+    'atTimestamp',
+    'everySeconds',
+    'everySecondsRange',
+    'everyVerseChorus',
+    'runUserSetup',
+    'runUserEvents',
+    'getCueList',
+  ];
+
+  const globals = new DanceAPI(api);
+  const code = interpreted + `return {${functionNames.join(',')}};`;
+
+  const params = [];
+  const args = [];
+  for (let k of Object.keys(globals)) {
+    params.push(k);
+    args.push(globals[k]);
+  }
+  params.push(code);
+  const ctor = function () {
+    return Function.apply(this, params);
+  };
+  ctor.prototype = Function.prototype;
+  const result = new ctor().apply(null, args);
```

Comment:
Worth de-duping this against `runLevel.js`?  I like the `return {${functionNames.join(',')}};` pattern, we could do the same for integration tests.

https://github.com/code-dot-org/dance-party/blob/dffdf04b6a1a7f28c0509660c2dde7faf85e888d/test/helpers/runLevel.js#L27-L38

Reply:
Ah I see, the `require('fs')` is needed in NodeJS, and `require('raw-loader!...)` is needed in the browser 😕.